ci-operator/step-registry/aws/provision/public-to-private/aws-provision-public-to-private-commands.sh

```diff
@@ -0,0 +1,162 @@
+	updatedCPMSCount=$(oc get controlplanemachineset/cluster -n openshift-machine-api -o=jsonpath='{.status.updatedReplicas}' || true)
+        if [[ ${readyCPMSCount} == "${DesiredCount}" ]] && [[ ${updatedCPMSCount} == "${DesiredCount}" ]]; then
+            echo "CPMS update finished."
+            break
+        fi
+    done
+    if [[ ${readyCPMSCount} != "${DesiredCount}" ]] || [[ ${updatedCPMSCount} != "${DesiredCount}" ]]; then
+	echo "Something wrong in the CPMS update:"
+        run_command "oc get machines -n openshift-machine-api"
+        run_command "oc get controlplanemachineset/cluster -n openshift-machine-api -o yaml"
+        return 1
+    fi
+}
+
+
+#Setting the Ingress Controller to private
+echo "Configuring DNS records to be published in the private zone"
+run_command "oc patch dnses.config.openshift.io/cluster --type=merge --patch='{\"spec\": {\"publicZone\": null}}'"
+
+echo "Setting the Ingress Controller to private"
+oc replace --force --wait --filename - <<EOF
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  namespace: openshift-ingress-operator
+  name: default
+spec:
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: Internal
+EOF
+
+
+# Remove the external load balancers from CPMS
+CPMS_PATCH=$(mktemp)
+cat >"${CPMS_PATCH}" <<EOF
+spec:
+  template:
+    machines_v1beta1_machine_openshift_io:
+      spec:
+        providerSpec:
+          value:
+            loadBalancers:
+            - name: ${INFRA_ID}-int
+              type: network
+EOF
+
+echo "Remove the external load balancers from CPMS"
+run_command "oc patch controlplanemachineset/cluster -n openshift-machine-api --type merge --patch-file ${CPMS_PATCH}"
+
+#Wait for CPMS update finished
+#masters was removed from the external LB, need to export proxy to access the cluster
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+else
+  echo "No proxy-conf found, exit now"
+  exit 1
+fi
+
+check_cpms_status
+
+
+#Restricting the API server to private by removing the external LB
+EXT_LB_ARN=$(aws --region ${REGION} elbv2 describe-load-balancers |jq --arg var "${INFRA_ID}" -r '.LoadBalancers[] | select(.LoadBalancerName == "\($var)-ext") | .LoadBalancerArn')
+
+if [[ -z ${EXT_LB_ARN} ]]; then
+  echo "Error: Fail to get the external LB ARN, exit"
+  exit 1
+fi
+
+echo "Delete the external load balancer"
+run_command "aws --region ${REGION} elbv2 delete-load-balancer --load-balancer-arn ${EXT_LB_ARN}"
+
+
+#Delete the API DNS entry in the public zone
+PUBLIC_ZONE_ID=$(aws route53 list-hosted-zones-by-name | jq --arg name "${BASE_DOMAIN}." -r '.HostedZones | .[] | select(.Name=="\($name)") | .Id' | awk -F / '{printf $3}')
+RECORD_SETS=$(aws route53 list-resource-record-sets --hosted-zone-id=${PUBLIC_ZONE_ID} --output json | jq --arg dns "api.${CLUSTER_NAME}.${BASE_DOMAIN}." '.ResourceRecordSets[] | select(.Name == "\($dns)")')
```

Comment:
Do we need to delete `*.apps` record manually ?

Reply:
The same as https://github.com/openshift/release/pull/59213#discussion_r1861943748